src/java.desktop/macosx/classes/sun/awt/CGraphicsDevice.java

```diff
@@ -280,6 +298,48 @@ private void initScaleFactor() {
         }
     }
 
+    // This method tries to create MTLGraphicsConfig
+    // if it fails, it tries to create CGLGraphicsConfig as a fallback
+    private void createMTLGraphicsConfig(final int displayID) {
+        this.config = MTLGraphicsConfig.getConfig(this, displayID, 0);
+
+        if (this.config != null) {
+            metalPipelineEnabled = true;
+        } else {
+            // Try falling back to OpenGL pipeline
+            if (MacOSFlags.isMetalVerbose()) {
+                System.out.println("Metal rendering pipeline initialization failed. Using OpenGL rendering pipeline.");
+            }
+
```

Comment:
Not sure I like this "side-effect" approach.
I think this method should return false if it fails and that get detected by the calller which then
decides what should happen next.